### ISSUE TITLE
New feature request to add custom data to output xml

### ISSUE THREAD

**Issue by varnk** [nunit/nunit#4280]
Would be great if inside my test code I could add some custom data to the output XML file so that I can include this on reports that I generate from the XML output using XSL.  
I am currently semi-accomplishing this by writing to the TestContext.Out stream.  However, this places the custom data that I write into the result output for the individual test method.  I would like something where I can write custom data that represents the entire run.

An example of this would be a set of name/value pairs, where I could write data items obtained from testing a HW device such as version, processor platform, part number of board, etc..

**Comment by yetibrain** [comment]
Some time ago i mentioned that parameter data passed to nunit console should be written in a specific header section within the output file in order for consumers that need the information how that console program has been executed with what parameters. Charly agreed. The need to pass specific custom information to the nunit resultfile, seems for me a feature that is worth beeing considered and realized.

**Comment by varnk** [comment]
> Some time ago i mentioned that parameter data passed to nunit console should be written in a specific header section within the output file in order for consumers that need the information how that console program has been executed with what parameters. Charly agreed. The need to pass specific custom information to the nunit resultfile, seems for me a feature that is worth beeing considered and realized.

I believe that parameter data passed to nunit-console is written to the test output XML file, which has come in handy for me.  However, I would also like to be able to add custom data pragmatically within the NUnit testing code too.

**Comment by yetibrain** [comment]
Absolutely. Indeed the result output file could be 'tuned/prepared' towards AI goals so that it can be used as a input file for nunit-console.exe in order to re-test failed tests. Infakt we do that since years , but using our own developed tools to achieve it the way we need it.
